ndonnx/_array.py

```diff
@@ -1,614 +1,443 @@
+    def __ne__(self, other) -> Array:  # type: ignore[override]
+        return _apply_op(self, other, std_ops.ne)
 
-    def min(self, axis: int | None = 0, keepdims: bool | None = False) -> ndx.Array:
-        """See :py:func:`ndonnx.min` for documentation."""
-        return ndx.min(self, axis=axis, keepdims=False)
+    ##################################################################
+    # __r*__ are needed for interacting with Python scalars          #
+    # (e.g. doing 1 + Array(...)). These functions are _NOT_ used to #
+    # dispatch between different `_TypedArray` subclasses.           #
+    ##################################################################
 
-    def all(self, axis: int | None = 0, keepdims: bool | None = False) -> ndx.Array:
-        """See :py:func:`ndonnx.all` for documentation."""
-        return ndx.all(self, axis=axis, keepdims=False)
+    __add__, __radd__ = _make_binary(std_ops.add)
+    __and__, __rand__ = _make_binary(std_ops.and_)
+    __floordiv__, __rfloordiv__ = _make_binary(std_ops.floordiv)
+    __ge__, _ = _make_binary(std_ops.ge)
+    __gt__, __rgt__ = _make_binary(std_ops.gt)
+    __le__, _ = _make_binary(std_ops.le)
+    __lshift__, __rlshift__ = _make_binary(std_ops.lshift)
+    __lt__, _ = _make_binary(std_ops.lt)
+    __matmul__, __rmatmul__ = _make_binary(std_ops.matmul)
+    __mod__, __rmod__ = _make_binary(std_ops.mod)
+    __mul__, __rmul__ = _make_binary(std_ops.mul)
+    __or__, __ror__ = _make_binary(std_ops.or_)
+    __pow__, __rpow__ = _make_binary(std_ops.pow)
+    __rshift__, __rrshift__ = _make_binary(std_ops.rshift)
+    __sub__, __rsub__ = _make_binary(std_ops.sub)
+    __truediv__, __rtruediv__ = _make_binary(std_ops.truediv)
+    __xor__, __rxor__ = _make_binary(std_ops.xor)
 
-    def any(self, axis: int | None = 0, keepdims: bool | None = False) -> ndx.Array:
-        """See :py:func:`ndonnx.any` for documentation."""
-        return ndx.any(self, axis=axis, keepdims=False)
+    def __abs__(self, /) -> Array:
+        data = self._tyarray.__abs__()
+        return Array._from_tyarray(data)
 
+    def __invert__(self, /) -> Array:
+        return Array._from_tyarray(~self._tyarray)
 
-class _Device:
-    # We would rather not give users the impression that their arrays
-    # are tied to a specific device when serializing an ONNX graph as
-    # such a concept does not exist in the ONNX standard.
+    def __neg__(self, /) -> Array:
+        return Array._from_tyarray(-self._tyarray)
 
-    def __str__(self):
-        return "ndonnx device"
+    def __pos__(self, /) -> Array:
+        return Array._from_tyarray(+self._tyarray)
 
-    def __eq__(self, other):
-        return type(other) is _Device
+    def __repr__(self) -> str:
+        value_repr = ", ".join(
+            [f"{k}: {v}" for k, v in self._tyarray.__ndx_value_repr__().items()]
+        )
+        shape = self._tyarray.shape
+        return f"array({value_repr}, shape={shape}, dtype={self.dtype})"
```

Comment:
The new print formatting gets a little long for constant arrays. Can we remove the shape for these?